src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,150 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.exceptions.InvalidDataException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static RegistrationService registrationService;
+    private User user;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        user = new User();
+        user.setLogin("loogiin");

```

Comment:
Best practice is to set up data in each test case: given (user), when, then.

Reply:
But we need some kind of basic user for most tests - and this is +3-4 lines of code in the method. In my tests, this is more than 30 lines of repeated code.

Can you please provide a link to the best practices?

Reply:
@BelitskyiAlexandr if you found that this way you can avoid a lot of code duplication and you're sure that this test user wouldn't change then it's ok to have it initialized in beforeeach as you've done